tests/units/anta_tests/__init__.py

```diff
@@ -3,30 +3,68 @@
 # that can be found in the LICENSE file.
 """Tests for anta.tests module."""
 
+from __future__ import annotations
+
 import asyncio
-from typing import Any
+from typing import TYPE_CHECKING, Any, Literal, TypedDict
+
+if TYPE_CHECKING:
+    import sys
 
-from anta.device import AntaDevice
+    from anta.device import AntaDevice
+    from anta.models import AntaTest
+    from anta.result_manager.models import AntaTestStatus
+
+    if sys.version_info >= (3, 11):
+        from typing import NotRequired
+    else:
+        from typing_extensions import NotRequired
+
+
+class UnitTestResult(TypedDict):
+    """Expected result of a unit test of an AntaTest subclass.
+    
+    For our AntaTest unit tests we expect only success, failure or skipped.
+    Never unset nor error.
+    """
 
+    result: Literal[AntaTestStatus.SUCCESS, AntaTestStatus.FAILURE, AntaTestStatus.SKIPPED]
+    messages: NotRequired[list[str]]
 
-def test(device: AntaDevice, data: dict[str, Any]) -> None:
+
+class AntaUnitTest(TypedDict):
+    """The parameters required for a unit test of an AntaTest subclass."""
+
+    inputs: NotRequired[dict[str, Any] | None]
```

Comment:
We should have either no key or a dict but not None
```suggestion
    inputs: NotRequired[dict[str, Any]]
```